Add suport for erofs in the kdump initrd

prudo1 opened the pull request:
This series adds support for erofs images in the kdump initrd. The feature was introduced in dracut with this [PR](https://github.com/dracut-ng/dracut-ng/pull/546) and will be included with dracut 104.

While working on the feature I got a little bit side tracked so this series also includes a few cleanups.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/rhkdump/kdump-utils/33)
<!-- Reviewable:end -->

coiby commented on .editorconfig:
```diff
@@ -20,13 +20,3 @@ space_redirects    = true
 [{mkfadumprd,mkdumprd,kdumpctl,kdump-lib.sh}]
 shell_variant = bash
-
-# Use dracut code style for *-module-setup.sh
```
Hi @prudo1,

Do I understand it correct that with this change, the kdump dracut modules won't adopt dracut code style?

coiby replied:
Oh, sorry for the notice. I didn't notice the editorconfig files in sub-folders until now.

coiby reviewed (changes requested):
> Reviewed 1 of 1 files at r1, 15 of 15 files at r2, 5 of 5 files at r3, 1 of 1 files at r4, 3 of 3 files at r5, 2 of 2 files at r6, 2 of 2 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, 1 of 1 files at r10, 1 of 1 files at r11, 1 of 1 files at r12, 1 of 1 files at r13, 2 of 2 files at r14, 3 of 3 files at r15, 2 of 2 files at r16, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/rhkdump/kdump-utils/33)* status: all files reviewed, 5 unresolved discussions (waiting on @prudo1)

___
*[`mkdumprd` line 16 at r9](https://reviewable.io/reviews/rhkdump/kdump-utils/33#-O4mVHdj8NVb09-UkndT:-O4mVHdj8NVb09-UkndU:bugbzu8) ([raw file](https://github.com/rhkdump/kdump-utils/blob/4921c36ec8c7dce8056a55c6c2f320537ae520c5/mkdumprd#L16)):*
> ```
> # shellcheck source=/dev/null
> . "$dracutbasedir"/dracut-functions.sh
> # shellcheck source-path=SCRIPTDIR
> ```

shellcheck still complains,
```
In mkdumprd line 10:
        . /etc/sysconfig/kdump
          ^------------------^ SC1091 (info): Not following: /etc/sysconfig/kdump was not specified as input (see shellcheck -x).

In mkdumprd line 17:
. /lib/kdump/kdump-lib.sh
  ^---------------------^ SC1091 (info): Not following: /lib/kdump/kdump-lib.sh was not specified as input (see shellcheck -x).


In mkdumprd line 18:
. /lib/kdump/kdump-logger.sh
  ^------------------------^ SC1091 (info): Not following: /lib/kdump/kdump-logger.sh was not specified as input (see shellcheck -x).
```
___
*[`mkfadumprd` line 13 at r10](https://reviewable.io/reviews/rhkdump/kdump-utils/33#-O4mVmb9082IUp3bePDA:-O4mVmb9082IUp3bePDB:b-nawcd0) ([raw file](https://github.com/rhkdump/kdump-utils/blob/dc32f191b22bf9395d54bfe34a0d13b7c9beb48b/mkfadumprd#L13)):*
> ```
> . "$dracutbasedir"/dracut-functions.sh
> # shellcheck source-path=SCRIPTDIR
> . /lib/kdump/kdump-lib.sh
> ```

Unforunately, shellcheck still makes the following complaints,
```
In mkfadumprd line 6:
        . /etc/sysconfig/kdump
          ^------------------^ SC1091 (info): Not following: /etc/sysconfig/kdump was not specified as input (see shellcheck -x).


In mkfadumprd line 13:
. /lib/kdump/kdump-lib.sh
  ^---------------------^ SC1091 (info): Not following: /lib/kdump/kdump-lib.sh was not specified as input (see shellcheck -x).


In mkfadumprd line 14:
. /lib/kdump/kdump-logger.sh
  ^------------------------^ SC1091 (info): Not following: /lib/kdump/kdump-logger.sh was not specified as input (see shellcheck -x).


In mkfadumprd line 24:
trap '
     ^-- SC2154 (warning): ret is referenced but not assigned.

```
___
*[`dracut/99kdumpbase/kdump.sh` line 8 at r5](https://reviewable.io/reviews/rhkdump/kdump-utils/33#-O4jUwQp-uUWkN-gpgVS:-O4jUwQp-uUWkN-gpgVT:btli0mn) ([raw file](https://github.com/rhkdump/kdump-utils/blob/82497dfc1d4cc7b28808dfb8dfccddee6516a8b4/dracut/99kdumpbase/kdump.sh#L8)):*
> ```Shell
> # shellcheck source=/dev/null
> . /lib/dracut-lib.sh
> # shellcheck source-path=SCRIPTDIR/../../
> ```

Unfortunately, `# shellcheck source=/dev/null` masks the next errors. If we remove it, we still the following errors,

```
In dracut/99kdumpbase/kdump.sh line 8:
. /lib/kdump-logger.sh
  ^------------------^ SC1091 (info): Not following: /lib/kdump-logger.sh was not specified as input (see shellcheck -x).


In dracut/99kdumpbase/kdump.sh line 9:
. /lib/kdump-lib-initramfs.sh
  ^-------------------------^ SC1091 (info): Not following: /lib/kdump-lib-initramfs.sh was not specified as input (see shellcheck -x).
```

So we need to explicit specify the paths,
```
# shellcheck source=SCRIPTDIR/../../kdump-logger.sh
. /lib/kdump-logger.sh
# shellcheck source=SCRIPTDIR/../../kdump-lib-initramfs.sh
. /lib/kdump-lib-initramfs.sh
```

In case the version matters, I use shellcheck-0.9.0 on F40 .
___
*[`dracut/99kdumpbase/module-setup.sh` line 21 at r5](https://reviewable.io/reviews/rhkdump/kdump-utils/33#-O4jXb8r2nHL2bGVdYOJ:-O4jXb8r2nHL2bGVdYOK:bl0obtx) ([raw file](https://github.com/rhkdump/kdump-utils/blob/82497dfc1d4cc7b28808dfb8dfccddee6516a8b4/dracut/99kdumpbase/module-setup.sh#L21)):*
> ```Shell
>     mkdir -p "$_DRACUT_KDUMP_NM_TMP_DIR"
> 
>     # shellcheck source-path=SCRIPTDIR/../../
> ```

Unfortunately, this doesn't work since we use absolute path.
___
*[`dracut/99kdumpbase/module-setup.sh` line 1110 at r5](https://reviewable.io/reviews/rhkdump/kdump-utils/33#-O4jYmZEDcveiLkGlgFT:-O4jYmZEDcveiLkGlgFU:bmzimjy) ([raw file](https://github.com/rhkdump/kdump-utils/blob/82497dfc1d4cc7b28808dfb8dfccddee6516a8b4/dracut/99kdumpbase/module-setup.sh#L1110)):*
> ```Shell
>     inst "/lib/kdump/kdump-lib-initramfs.sh" "/lib/kdump-lib-initramfs.sh"
>     inst "/lib/kdump/kdump-logger.sh" "/lib/kdump-logger.sh"
>     # shellcheck disable=SC2154
> ```

`# shellcheck disable=SC2154` is not needed here since the early rule also applies here.


<!-- Sent from Reviewable.io -->

coiby commented:
Hi @prudo1,

This PR looks good to me except for some issues about shellcheck.

prudo1 commented:
Hi @coiby,

unfortunately github doesn't allow to reply in-line so ...

> > Reviewed 1 of 1 files at r1, 15 of 15 files at r2, 5 of 5 files at r3, 1 of 1 files at r4, 3 of 3 files at r5, 2 of 2 files at r6, 2 of 2 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, 1 of 1 files at r10, 1 of 1 files at r11, 1 of 1 files at r12, 1 of 1 files at r13, 2 of 2 files at r14, 3 of 3 files at r15, 2 of 2 files at r16, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/rhkdump/kdump-utils/33)_ status: all files reviewed, 5 unresolved discussions (waiting on @prudo1)
> 
> _[`mkdumprd` line 16 at r9](https://reviewable.io/reviews/rhkdump/kdump-utils/33#-O4mVHdj8NVb09-UkndT:-O4mVHdj8NVb09-UkndU:bugbzu8) ([raw file](https://github.com/rhkdump/kdump-utils/blob/4921c36ec8c7dce8056a55c6c2f320537ae520c5/mkdumprd#L16)):_
> 
> > ```
> > # shellcheck source=/dev/null
> > . "$dracutbasedir"/dracut-functions.sh
> > # shellcheck source-path=SCRIPTDIR
> > ```
> 
> shellcheck still complains,
> 
> ```
> In mkdumprd line 10:
>         . /etc/sysconfig/kdump
>           ^------------------^ SC1091 (info): Not following: /etc/sysconfig/kdump was not specified as input (see shellcheck -x).
> 
> In mkdumprd line 17:
> . /lib/kdump/kdump-lib.sh
>   ^---------------------^ SC1091 (info): Not following: /lib/kdump/kdump-lib.sh was not specified as input (see shellcheck -x).
> 
> 
> In mkdumprd line 18:
> . /lib/kdump/kdump-logger.sh
>   ^------------------------^ SC1091 (info): Not following: /lib/kdump/kdump-logger.sh was not specified as input (see shellcheck -x).
> ```

Those are all only hints that you need to call `shellcheck` with option `-x` for it to follow those files. When you do that the warnings goes away. Same for most of the other warnings you mentioned.
 
> _[`mkfadumprd` line 13 at r10](https://reviewable.io/reviews/rhkdump/kdump-utils/33#-O4mVmb9082IUp3bePDA:-O4mVmb9082IUp3bePDB:b-nawcd0) ([raw file](https://github.com/rhkdump/kdump-utils/blob/dc32f191b22bf9395d54bfe34a0d13b7c9beb48b/mkfadumprd#L13)):_
> 
> > ```
> > . "$dracutbasedir"/dracut-functions.sh
> > # shellcheck source-path=SCRIPTDIR
> > . /lib/kdump/kdump-lib.sh
> > ```
> 
> Unforunately, shellcheck still makes the following complaints,

[...]

> In mkfadumprd line 24:
> trap '
>      ^-- SC2154 (warning): ret is referenced but not assigned.
> ```

Ouch, this one was masked by using `-x` as ret is defined in the `kdump-logger.sh`...
I've disabled the warning here and in `mkdumprd`.
I'm not entirely sure why it doesn't show in `kdumpctl`. Shall I disable it there, too?
 
[...]

> 
> In case the version matters, I use shellcheck-0.9.0 on F40 .

I'm using the same version. Although, when I created the series I was still on F39. But IIRC the shellcheck version was the same.

> 
> _[`dracut/99kdumpbase/module-setup.sh` line 21 at r5](https://reviewable.io/reviews/rhkdump/kdump-utils/33#-O4jXb8r2nHL2bGVdYOJ:-O4jXb8r2nHL2bGVdYOK:bl0obtx) ([raw file](https://github.com/rhkdump/kdump-utils/blob/82497dfc1d4cc7b28808dfb8dfccddee6516a8b4/dracut/99kdumpbase/module-setup.sh#L21)):_
> 
> > ```shell
> >     mkdir -p "$_DRACUT_KDUMP_NM_TMP_DIR"
> > 
> >     # shellcheck source-path=SCRIPTDIR/../../
> > ```
> 
> Unfortunately, this doesn't work since we use absolute path.

Right, I misunderstood the shellcheck documentation. I thought that it would always only use the filename and attach it to `SCRIPTDIR`. Do you know if there is a way to tell `shellcheck` to print out the absolute path it uses as source? That would be handy when adding those statements so you can verify that `shellcheck` uses the right file.

Anyway, I've dropped the use of `source-path` and explicitly set `source` every time. This should also solve the problem with `source=/dev/null` hiding problems.

> 
> _[`dracut/99kdumpbase/module-setup.sh` line 1110 at r5](https://reviewable.io/reviews/rhkdump/kdump-utils/33#-O4jYmZEDcveiLkGlgFT:-O4jYmZEDcveiLkGlgFU:bmzimjy) ([raw file](https://github.com/rhkdump/kdump-utils/blob/82497dfc1d4cc7b28808dfb8dfccddee6516a8b4/dracut/99kdumpbase/module-setup.sh#L1110)):_
> 
> > ```shell
> >     inst "/lib/kdump/kdump-lib-initramfs.sh" "/lib/kdump-lib-initramfs.sh"
> >     inst "/lib/kdump/kdump-logger.sh" "/lib/kdump-logger.sh"
> >     # shellcheck disable=SC2154
> > ```
> 
> `# shellcheck disable=SC2154` is not needed here since the early rule also applies here.

Thanks! Dropped it.

prudo1 commented:
Added changes requested by @coiby

liutgnu reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/rhkdump/kdump-utils/33)* status: 8 of 19 files reviewed, 6 unresolved discussions (waiting on @coiby and @prudo1)

___
*[`mkdumprd` line 70 at r29](https://reviewable.io/reviews/rhkdump/kdump-utils/33#-O5HM4hXEhvOZCXzvEBU:-O5HM4hXEhvOZCXzvEBV:bu8i6a1) ([raw file](https://github.com/rhkdump/kdump-utils/blob/123a3c5a31e8f5c37eec5503580f3d3e77053545/mkdumprd#L70)):*
> ```
> 	done
> 
> 	dracut --list-modules | grep -x -q "${_args[@]}"
> ```

I guess it will have a problem same as commit ("98087d78eda Use "grep -q <<< $(cmd)" instead of "cmd | grep -q""), could we use "<<<" instead?


<!-- Sent from Reviewable.io -->

prudo1 commented:
fixed the problem pointed out by @liutgnu

liutgnu reviewed (approved):
LGTM
